File: 자료구조/스택/백준_10828_스택/스택.cs
using System;
using System.Text;
using static System.Console;

class Program
{
    public static void Main(string[] args)
    {
        StringBuilder ans = new StringBuilder();
        int[] arr = new int[100001];
        int n = int.Parse(ReadLine());
        int f = 0;
        while (n-- > 0)
        {
            string[] s = ReadLine().Split(' ');
            if (s[0] == "push")
            {
                int num = int.Parse(s[1]);
                arr[f] = num;
                f++;
            }
            else if (s[0] == "pop")
            {
                if (f == 0) ans.Append(-1 + "\n");
                else
                {
                    ans.Append(arr[f - 1] + "\n");
                    arr[f - 1] = 0;
                    f--;
                }
            }
            else if (s[0] == "size") ans.Append(f + "\n");
            else if (s[0] == "empty")
            {
                if(f == 0) ans.Append(1 + "\n");
                else ans.Append(0 + "\n");
            }
            else if (s[0] == "top")
            {
                if(f== 0) ans.Append(-1 + "\n");
                else ans.Append(arr[f-1] + "\n");
            }
        }
        WriteLine(ans.ToString());
    }
}

File: 자료구조/스택/백준_10828_스택/스택.cpp
#include <iostream>
#include <cstring>
using namespace std;

int arr[10001];

int main() {
    int n = 0;
    scanf("%d", &n);
    char s[10];
    int f = 0;

    while(n--){
        scanf("%s", &s);
        if(strcmp(s, "push") == 0){
            int num = 0;
            scanf("%d", &num);
            arr[f] = num;
            f++;
        }
        else if(strcmp(s, "pop") == 0){
            if(f == 0) printf("-1\n");
            else{
                printf("%d\n", arr[f-1]);
                arr[f-1] = 0;
                f--;
            }
        }
        else if(strcmp(s, "size") == 0) printf("%d\n", f);
        else if(strcmp(s, "empty") == 0){
            if(f == 0) printf("1\n");
            else printf("0\n");
        }
        else if(strcmp(s, "top") == 0){
            if(f == 0) printf("-1\n");
            else printf("%d\n", arr[f-1]);
        }
    }


}

